File: Components/App.Logic/Models/SearchAggregationModel.cs
using System;
using System.Collections.Generic;

namespace App.Logic.Models
{
    public class SearchAggregationModel
    {
        public DateTime SearchDate { get; set; }

        public Guid SearchIdentifier { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string MiddleName { get; set; }

        public int TotalOccurrences { get; set; }

        public IEnumerable<SearchResultsModel> Results { get; set; }
    }
}

File: Web/app.client/App.App/src/app/models/SearchAggregationModel.ts
export interface SearchAggregationModel {
  searchDate: string;
  searchIdentifier: string;
  firstName: number;
  lastName: string;
  middleName: string;
  totalOccurrences: number;
  results: SearchResultModel[];
}

export interface SearchResultModel {
  articleIdentifier: string;
  articleName: string;
  data: SearchDataModel[];
}

export interface SearchDataModel {
  searchText: string;
  occurrences: number;
}
